### ISSUE TITLE
Toolbar button not working

### ISSUE THREAD

**Issue by jonathanKingston** [mozilla/testpilot-containers#679]
From @moi12518 in #670

I've been searching through the issues list for something similar to my problem and this is the closest I've gotten by far. For quite some time, I've had one issue with Containers. The icon on the toolbar works and allows me to set up Containers for only a few sessions after Containers are enabled before it stops working entirely. This isn't drastically impacting my usage, as after I recognized the pattern of 'enable Containers, a few sessions pass, and then the icon is useless,' I disabled and re-enabled Containers to create all the Containers I could see myself using and haven't needed to add any since. However, I can imagine that someone else could also have this problem and it be a serious one, so I'm speaking my bit now (as I finally found out I could report problems). 

![image](https://user-images.githubusercontent.com/30208004/28245498-2146f6aa-69bd-11e7-887d-b869d674139d.png)

Not sure if that worked, so I'll describe the picture I tried to put in just in case. The icon looks normal when I don't to anything to it, but when I click it, only a white dropdown arrow appears. There is no popup/bubble thing at all. This is replicable and happens every time I try to click the icon in every session. I have AdBlock Plus, uBlock Origin, and Ghostery, so it's possible one of them or a combination thereof is the source of the problem. (There's also Blur which just appeared one day in a Firefox update, so if it's the problem then I'll be surprised.) I've considered that having a Firefox Persona may be important, but it was a brief consideration.

I know this isn't the same problem, but it's similar enough that they might be a bit related. This issue has existed for months and had I realized I could have reported it sooner, I would have.

**Comment by jonathanKingston** [comment]
@moi12518 would you be able to provide details of what is in the contents of about:support page? This should provide what platform you are on and extensions available.
Did this item just stop working?
How were you able to configure this before?

**Comment by moi12518** [comment]
Name: Firefox
Version: 52.2.1
Build ID: 20170627155318
Update Channel: esr
User Agent: Mozilla/5.0 (Windows NT 6.0; WOW64; rv:52.0) Gecko/20100101 Firefox/52.0
OS: Windows_NT 6.0
Multiprocess Windows: 3/3 (Enabled by default)
Safe Mode: false

Crash Reports for the Last 3 Days, All Crash Reports (including 2 pending crashes in the given time range)
d01965b2-e23e-4f31-8f40-9deeb71b73c4
	7/16/17	1:29 PM
35dac2a4-2726-4e61-97b1-e6d8bdf67d16
	7/16/17	10:30 AM

![image](https://user-images.githubusercontent.com/30208004/28348770-9fa5ed98-6bf3-11e7-88f1-40c4a4a3ffa2.png)
![image](https://user-images.githubusercontent.com/30208004/28348787-bcef56d2-6bf3-11e7-8ab5-cd0d125cee82.png)
![image](https://user-images.githubusercontent.com/30208004/28348795-cfce6658-6bf3-11e7-8dbf-75e0f66c7aad.png)
![image](https://user-images.githubusercontent.com/30208004/28348801-dbe7678c-6bf3-11e7-811b-33fbf4471379.png)
![image](https://user-images.githubusercontent.com/30208004/28348810-ea3e5e6c-6bf3-11e7-923a-98c7876fc31b.png)
![image](https://user-images.githubusercontent.com/30208004/28348818-fb370ea8-6bf3-11e7-95ec-fb0d534fa81d.png)
![image](https://user-images.githubusercontent.com/30208004/28348833-14d32338-6bf4-11e7-95f6-3e1eda68a52d.png)
![image](https://user-images.githubusercontent.com/30208004/28348864-3a7cd4d0-6bf4-11e7-8de6-0f81800f8c36.png)
![image](https://user-images.githubusercontent.com/30208004/28348942-b6075e9a-6bf4-11e7-8a93-53e348587fbd.png)
![image](https://user-images.githubusercontent.com/30208004/28349046-4cdf3644-6bf5-11e7-9281-d70f48219bbf.png)
![image](https://user-images.githubusercontent.com/30208004/28349110-9ce1fd34-6bf5-11e7-9eaa-3d3bb50e1ed1.png)

I don't really know all that much about computers and trying to copy and paste the text on the page didn't work well, so have a bunch of screenshots. 

After a few sessions after enabling Containers, the icon stops working. I don't remember specifics of exactly how many sessions it takes, but I remember the number varying a bit. It just stops working; one session it's fine, the next all I get then on is the arrow. There are no warnings.

I'm not sure quite what you mean by 'configure'; assuming you mean 'set up Containers,' the dropdown worked and I was able to create separate shells. There wasn't anything unusual about it. If by 'configure' you mean 'dodge the problem,' I just followed the time-old advice "have you tried turning it off and back on again?" Sorry, I'm not a computer person.

**Comment by kjozwiak** [comment]
@moi12518 thanks for the info! I really appreciate it. Next time you run into the above issue, could you please check the Console Browser and paste any errors you see into this bug? You can access the browser console via:

* Select the` Hamburger Menu` on the right hand side
* Select `Developer` and than select `Browser Console`

It also looks like the @moi12518 is using ESR. I wonder if that could be causing some issues.

In the meantime, I'll try getting some STR using ESR and the add-ons listed above.

**Comment by kjozwiak** [comment]
@jonathanKingston, this appears to be an ESR issue. Both @moi12518 and @klint from #682 are using the ESR branch. You can reproduce the issue pretty easily without any add-ons using the following STR:

* download and install the latest esr branch [here](https://www.mozilla.org/en-US/firefox/organizations/all/). (latest build is fx52.2.1esr)
* download the latest [test pilot](https://testpilot.firefox.com/experiments) and [containers experiment](https://testpilot.firefox.com/experiments/containers)
* once installed, simply click on the `container button` and you'll notice the issue occurring

When clicking on the `container button` several times, you'll get the following errors in the browser console:

```
TypeError: browser.contextualIdentities is undefined popup.js:188:7

Cannot send function call result: other side closed connection  ExtensionParent.jsm:460
	call/reply resource://gre/modules/ExtensionParent.jsm:460:9
	call/< resource://gre/modules/ExtensionParent.jsm:482:11
	Handler.prototype.process resource://gre/modules/Promise-backend.js:932:23
	this.PromiseWalker.walkerLoop resource://gre/modules/Promise-backend.js:813:7
	bound  self-hosted:913:17
	bound bound  self-hosted:913:17
	this.PromiseWalker.scheduleWalkerLoop/< resource://gre/modules/Promise-backend.js:747:11

Error: Popup destroyed ext-utils.js:124:39
```

Results: (seems like it's broken for a while now under ESR)

* fx52.2.1esr, buildid: 20170627155318, changeset: 512efd480dac - Reproduced
* fx52.2.0esr, buildid: 20170607123825, changeset: f68e0d98a22a - Reproduced
* fx52.1.2esr, buildid: 20170517122419, changeset: 0dc7c28f2064 - Reproduced
* fx52.1.1esr, buildid: 20170504112025, changeset: 120111e65bc4 - Reproduced
* fx52.0esr, buildid: 20170303022339, changeset: 15d9d940341f - Reproduced

**Comment by jonathanKingston** [comment]
@kjozwiak thanks for the debugging, sorry missed they were all 52. I was a little worried that we had it elsewhere.

I raised #686 to consider a plan to support these users. I'm going to close this for now and I will put the advice from other bugs I listed there too so that we don't duplicate this.

**Comment by DHowett** [comment]
I am seeing this behavior in 56.0a1. Should I collect logs and file a new instance of this bug?

**Comment by DHowett** [comment]
Sorry for the noise; a Nightly update resolved this.

**Comment by sharuzzaman** [comment]
Hi,

Got the same issue with 52.4.0 (32-bit)

This site does not mentioned that it requires Firefox 53+ . See: https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/

Please block the installation if it is not supported for ESR.

Thanks.

**Comment by groovecoder** [comment]
This is fixed in [the 4.0.3 release](https://github.com/mozilla/multi-account-containers/releases/tag/4.0.3) (by setting the strict min version to 53). It's uploaded to AMO in the review queue now.
